src/monitors/MonitorOptions.v3

```diff
@@ -44,21 +44,59 @@ component MonitorOptions {
 			var engine = Engine.new();
 			engine.extensions = Extension.set.all;
 			var tiering_override = if(SpcTuning.compileWhammModules, ExecuteOptions.getModeOrDefault("jit"), Execute.tiering);
-			var result = engine.loadWasmFileWithTiering(name, tiering_override); // TODO: limits
-			match (result) {
-				Ok(m) => {
-					monitors.put(WhammMonitor.new(m));
-					return;
+
+			// Split by '+'
+			var s = 0, pos = s;
+			var whammModuleName = "";
+			var otherModuleNames = Vector<string>.new();
+			while (pos < name.length) {
+				match (name[pos]) {
+					'+' => {
+						var name = Arrays.range(name, s, pos);
+
+						// First module is the whamm module, otherwise it's a library to link
+						if (whammModuleName.length == 0) {
+							whammModuleName = name;
+						} else {
+							otherModuleNames.put(name);
+						}
+						s = pos + 1;
+						pos++;
+					}
+					_ => pos++;
 				}
-				FileNotFound => return ErrorBuilder.new()
-					.puts("wizeng: could not load monitor file ")
-					.puts(name)
-					.ln()
-					.exit<void>();
-				ParseError(code, path, section, error_index, error_pos, error_msg) => return ErrorBuilder.new()
-					.render(code, path, section, error_index, error_pos, error_msg)
-					.exit<void>();
 			}
+			// If whamm module hasn't been set yet, we don't have any libraries to link
+			var finalName = Arrays.range(name, s, pos);
+			if (!Strings.endsWith(finalName, ".wasm")) {
+				ErrorBuilder.new()
```

Comment:
I feel like this is a good time and bite the bullet to feed in an `ErrorGen` to `parse()` above and feed it here. Then it's the caller's responsibility to pass one in and report errors.